engine/gui/src/gui.cpp

```diff
@@ -1409,7 +1409,7 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
     {
         bool no_clipping = n->m_ClipperIndex == INVALID_INDEX;
 
-        return opacity == 0.0f && no_clipping;
+        return !n->m_Node.m_IsVisible || opacity == 0.0f && no_clipping;
```

Comment:
Mixing negated values with OR's and AND can be a bit confusing.
I suggest refactoring this to `IsVisible`:
```
static inline bool IsVisible(InternalNode* n, float opacity)
{
    bool use_clipping = n->m_ClipperIndex != INVALID_INDEX;
    
    return n->m_Node.m_IsVisible && (opacity != 0.0f || use_clipping);
}
```